core/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -173,6 +173,7 @@
         this.timeout = timeout;
         queryShardContext = indexService.newQueryShardContext(request.shardId().id(), searcher.getIndexReader(), request::nowInMillis);
         queryShardContext.setTypes(request.types());
+        queryBoost = request.indexBoost();
```

Comment:
I think that we can remove the query boost setter and make the instance member final.